exercise/exercise_models.py

```diff
@@ -694,7 +695,16 @@ def grade(self, request, submission, no_penalties=False, url_name="exercise"):
         """
         Loads the exercise feedback page.
         """
-        language = get_language()
+        # Get the language from the submission
+        try:
+            aplus_data_index = [i[0] for i in submission.submission_data].index("__aplus__")
+            aplus_data_string = submission.submission_data[aplus_data_index][-1]
+            aplus_data = json.loads(aplus_data_string)
+            language = aplus_data.get("lang",  get_language())
```

Comment:
For information. It's beneficial to think when things are evaluated. Here, the `get_language` is evaluated always before `aplus_data.get`. It should be cheap, but as it's a function call we can't be sure when reading the code (function calls are expected to do processing while attribute resolution should be "cheap").

Hence, things like this might be beneficial to write `language = aplus_data.get('lang') or get_language()`. The `or` makes it so that the first part is evaluated and only if that is false (None or empty string), then the latter part is evaluated.

In addition, try to use `'` for for dictionary key strings (`'lang'` vs. `"lang"`). That wey the quotes can be used to indicate the strings purpose/type for the reader. There is a bit more in the style guide about this, once we get that done.

Reply:
Good to know I thought it wouldn't be evaluated if lang is found, but I guess it's logical that the arguments are evaluated before the function is called.